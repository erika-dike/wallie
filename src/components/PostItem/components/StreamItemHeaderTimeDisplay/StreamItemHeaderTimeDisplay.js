import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import { OverlayTrigger, Tooltip } from 'react-bootstrap';

import { getFullTime } from '../../../../utils';

import './StreamItemHeaderTimeDisplay.css';


const StreamItemHeaderTimeDisplay = ({ date_created }) => {
  const getTimeTooltip = time =>
    <Tooltip id="time-tooltip">{getFullTime(time)}</Tooltip>;

  return (
    <small className="time">
      <OverlayTrigger
        overlay={getTimeTooltip(date_created)}
        placement="top"
        delayShow={200}
        delayHide={150}
      >
        <a className="post-timestamp">
          <span className="timestamp">
            {moment(date_created).fromNow(true)}
          </span>
          <span style={{ display: 'none' }}>
            {moment(date_created).fromNow()}
          </span>
        </a>
      </OverlayTrigger>
    </small>
  );
};

StreamItemHeaderTimeDisplay.propTypes = {
  date_created: PropTypes.string.isRequired,
};

export default StreamItemHeaderTimeDisplay;
