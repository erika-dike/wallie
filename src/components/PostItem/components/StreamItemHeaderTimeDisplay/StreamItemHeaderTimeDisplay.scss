.time {
    font-size: 14px;
    color: #657786;
}

.post {
    .time {
        color: #657786;
    }
}