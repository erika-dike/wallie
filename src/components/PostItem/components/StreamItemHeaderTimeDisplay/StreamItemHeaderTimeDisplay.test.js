import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import StreamItemHeaderTimeDisplay from './StreamItemHeaderTimeDisplay';


describe('StreamItemHeaderTimeDisplay component test suite', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      date_created: '2017-05-10T11:41:08.735591Z',
    };
  });

  describe('Base test', () => {
    beforeEach(() => {
      wrapper = shallow(<StreamItemHeaderTimeDisplay {...props} />);
    });

    it('renders without crashing', () => {
      expect(wrapper).toBeDefined();
    });

    it('has 1 prop', () => {
      expect(Object.keys(wrapper.instance().props)).toHaveLength(1);
    });

    it('renders correct snapshot', () => {
      const tree = renderer.create(<StreamItemHeaderTimeDisplay />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
