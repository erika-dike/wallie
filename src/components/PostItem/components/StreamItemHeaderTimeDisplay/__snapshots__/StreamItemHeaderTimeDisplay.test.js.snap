exports[`StreamItemHeaderTimeDisplay component test suite Base test renders correct snapshot 1`] = `
<small
  className="time">
  <a
    className="post-timestamp"
    onBlur={[Function]}
    onClick={null}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}>
    <span
      className="timestamp">
      a month
    </span>
    <span
      style={
        Object {
          "display": "none",
        }
      }>
      a month ago
    </span>
  </a>
</small>
`;
